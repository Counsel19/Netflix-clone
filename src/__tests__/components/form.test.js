/* eslint-disable testing-library/no-node-access */
import React from "react";
import { render, screen } from "@testing-library/react";
import { Form } from "../../components";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";

describe("<Form />", () => {
  it("renders the <Form /> with populated data", () => {
    const { container } = render(
      <MemoryRouter>
        <Form>
          <Form.Title>Sign In</Form.Title>

          <Form.Base>
            <Form.Input
              type="email"
              placeholder="Email Address"
              onChange={() => {}}
            />
            <Form.Input
              type="password"
              placeholder="Password"
              onChange={() => {}}
            />

            <Form.Submit disabled type="submit">
              Sign In
            </Form.Submit>
          </Form.Base>

          <Form.Text>
            New to Netflix? <Form.Link to="/signup">Sign up now</Form.Link>
          </Form.Text>

          <Form.TextSmall>
            This page is protected by Google reCAPTCHA to ensure you're not a
            bot. Learn more.
          </Form.TextSmall>
        </Form>
      </MemoryRouter>
    );

    expect(
      screen.getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();
    expect(screen.getByText(/sign up now/i)).toBeTruthy();
    expect(screen.getAllByText(/sign in/i)).toBeTruthy();
    expect(
      screen.getByRole("button", { name: /sign in/i }).disabled
    ).toBeTruthy();
    expect(screen.getByPlaceholderText(/email address/i)).toBeTruthy();
    expect(screen.getByPlaceholderText(/Password/i)).toBeTruthy();
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Form /> with an error", () => {
    const { container } = render(
      <Form>
        <Form.Error>Your Emaill address is currently being used</Form.Error>
        <Form.Submit type="submit">
          Sign In
        </Form.Submit>
      </Form>
    );

    expect(screen.getByText('Your Emaill address is currently being used')).toBeTruthy();
    // eslint-disable-next-line testing-library/prefer-presence-queries
    expect(screen.getByRole('button', {name: /sign in/i}).disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot()
  });
});
