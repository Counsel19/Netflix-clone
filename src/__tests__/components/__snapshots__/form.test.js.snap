// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with an error 1`] = `
<div
  class="form__Container-sc-az6gvf-0 hDhTub"
>
  <div
    class="form__Error-sc-az6gvf-1 kELLgU"
  >
    Your Emaill address is currently being used
  </div>
  <button
    class="form__Submit-sc-az6gvf-8 kFKbVY"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-az6gvf-0 hDhTub"
>
  <h1
    class="form__Title-sc-az6gvf-3 fuxRbP"
  >
    Sign In
  </h1>
  <form
    class="form__Base-sc-az6gvf-2 joblgE"
  >
    <input
      class="form__Input-sc-az6gvf-7 pdnTS"
      placeholder="Email Address"
      type="email"
    />
    <input
      class="form__Input-sc-az6gvf-7 pdnTS"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-az6gvf-8 kFKbVY"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-az6gvf-4 xUyEe"
  >
    New to Netflix? 
    <a
      class="form__Link-sc-az6gvf-6 cPPpZa"
      href="/signup"
    >
      Sign up now
    </a>
  </p>
  <p
    class="form__TextSmall-sc-az6gvf-5 ixxGsK"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
